src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -53,58 +53,119 @@ end
 )
     return u * dot(occ_a, occ_b)
 end
-@inline function local_interaction(fs::CompositeFS, u, occs)
-    return _interactions(fs.components, u, occs)
+@inline local_interaction(::SingleComponentFockAddress, ::SingleComponentFockAddress, 
+    ::Nothing,_,_) = 0
+"""
+    nearest_neighbour_interaction(onr1::SVector, onr2::SVector, Δ, geometry::CubicGrid, map1::ModeMap)
+
+Calculate the nearest neighbour interaction ``Δ \\sum_{⟨i,j⟩} n_{↑, i} n_{↓, j}`` for an occupation
+number representation `onr1` and `onr2` of the two single-component Fock states. For a 
+multi-component Fock state, return the eigenvalue of
+
+```math
+\\frac{1}{2}\\sum_{⟨i,j⟩, σ, τ} Δ_{σ,τ} a^†_{σ,i}a^†_{τ,j}a_{τ,j}a_{σ,i} ,
+```
+
+where `Δ::SMatrix` is a symmetric matrix of interaction constants, `i, j` are the mode indices,
+and `σ`, `τ` are component indices.
+
+See also [`BoseFS`](@ref), [`FermiFS`](@ref), [`CompositeFS`](@ref).
+"""
+@inline function nearest_neighbour_interaction(onr1::SVector, onr2::SVector, 
+    Δ, geometry::CubicGrid{D,S,B}, map1::ModeMap
+    ) where {D,S,B}
+    N1 = length(map1)
+    ext_result = 0
+    for i in 1:N1
+        for j in 1:D
+            occ_mode1 = map1[i].mode
+            neigh = neighbor_site(geometry, occ_mode1, j)
+            if !iszero(neigh)
+                ext_result += onr1[occ_mode1] * onr2[neigh]
+            end
+        end
+    end
+    return Δ*ext_result
+end
+@inline nearest_neighbour_interaction(::SVector, ::SVector, ::Nothing, ::CubicGrid, ::ModeMap) = 0
```

Comment:
```suggestion
@inline nearest_neighbor_interaction(::SVector, ::SVector, ::Nothing, ::CubicGrid, ::ModeMap) = 0
```
Unfortunately, we use American spelling in the code 😄